corehq/apps/es/es_query.py

```diff
@@ -249,7 +249,7 @@ def scroll(self):
 
     @property
     def _filters(self):
-        return self.es_query['query']['filtered']['filter']['and']
+        return self.es_query['query']['bool']['must']
```

Comment:
Shouldn't the filters go in the `filter` block of the `bool` query, not `must`, since they aren't used for scoring?

Reply:
@esoergel If I good understand how the ```bool``` query works in new version then ```bool must``` is correct and should work the same as we have previously. In ES 6.4 ```filtered```, ```and``` and ```or``` filters were deprecated and was replaced by bool query. https://www.elastic.co/guide/en/elasticsearch/reference/6.4/query-dsl-bool-query.html

Reply:
From that docs page:
> **must** The clause (query) must appear in matching documents and will contribute to the score.
> **filter** The clause (query) must appear in matching documents. However unlike must the score of the query will be ignored. Filter clauses are executed in filter context, meaning that scoring is ignored and clauses are considered for caching.

I take that to mean both will work, in that they'll return the same results, but the difference is that `must` will contribute to scoring, and those clauses won't be cached.  The `filter` context is used for filtering structured data.  Here's [the docs page on the difference](https://www.elastic.co/guide/en/elasticsearch/reference/6.4/query-filter-context.html).

The [docs for migrating](https://www.elastic.co/guide/en/elasticsearch/reference/6.4/query-dsl-filtered-query.html) from `filtered` to `bool` also say to move filters to the `filter` block.